demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -1,1081 +1,557 @@
-    {
-        LogInfo( ( "Creating a TLS connection to %s:%u.\r\n", pcHostName, port ) );
-        /* Attempt to create a mutually authenticated TLS connection. */
-        xNetworkStatus = TLS_Socket_Connect( pxNetworkContext,
-                                             pcHostName, port,
-                                             pxNetworkCredentials,
-                                             sampleazureiotTRANSPORT_SEND_RECV_TIMEOUT_MS,
-                                             sampleazureiotTRANSPORT_SEND_RECV_TIMEOUT_MS );
-
-        if( xNetworkStatus != eTLSTransportSuccess )
-        {
-            /* Generate a random number and calculate backoff value (in milliseconds) for
-             * the next connection retry.
-             * Note: It is recommended to seed the random number generator with a device-specific
-             * entropy source so that possibility of multiple devices retrying failed network operations
-             * at similar intervals can be avoided. */
-            xBackoffAlgStatus = BackoffAlgorithm_GetNextBackoff( &xReconnectParams, configRAND32(), &usNextRetryBackOff );
-
-            if( xBackoffAlgStatus == BackoffAlgorithmRetriesExhausted )
-            {
-                LogError( ( "Connection to the IoT Hub failed, all attempts exhausted." ) );
-            }
-            else if( xBackoffAlgStatus == BackoffAlgorithmSuccess )
-            {
-                LogWarn( ( "Connection to the IoT Hub failed [%d]. "
-                           "Retrying connection with backoff and jitter [%d]ms.",
-                           xNetworkStatus, usNextRetryBackOff ) );
-                vTaskDelay( pdMS_TO_TICKS( usNextRetryBackOff ) );
-            }
-        }
-    } while( ( xNetworkStatus != eTLSTransportSuccess ) && ( xBackoffAlgStatus == BackoffAlgorithmSuccess ) );
-
-    return xNetworkStatus == eTLSTransportSuccess ? 0 : 1;
-}
-/*-----------------------------------------------------------*/
-
-/*
- * @brief Create the task that demonstrates the AzureIoTHub demo
- */
-void vStartDemoTask( void )
-{
-    /* This example uses a single application task, which in turn is used to
-     * connect, subscribe, publish, unsubscribe and disconnect from the IoT Hub */
-    xTaskCreate( prvAzureDemoTask,         /* Function that implements the task. */
-                 "AzureDemoTask",          /* Text name for the task - only used for debugging. */
-                 democonfigDEMO_STACKSIZE, /* Size of stack (in words, not bytes) to allocate for the task. */
-                 NULL,                     /* Task parameter - not used in this case. */
-                 tskIDLE_PRIORITY,         /* Task priority, must be between 0 and configMAX_PRIORITIES - 1. */
-                 NULL );                   /* Used to pass out a handle to the created task - not used in this case. */
-}
-/*-----------------------------------------------------------*/
+/* Copyright (c) Microsoft Corporation. All rights reserved. */

```

Comment:
This is a split from the original sample_azure_iot_pnp.c, containing only the code that deals with the implementation of the PnP template and the functions defined in the interface header sample_azure_iot_pnp.h